src/vs/workbench/common/editor/untitledTextEditorModel.ts

```diff
@@ -63,6 +63,11 @@ export class UntitledTextEditorModel extends BaseTextEditorModel implements IUnt
 		this._register(this.configurationService.onDidChangeConfiguration(e => this.onConfigurationChange()));
 	}
 
+	public registerDirtyListener(): void {
+		// For editData
```

Comment:
Why are you doing this here? Generally we should avoid having specific logic like this in a more general-use class.

Reply:
And FYI - if you're ever making changes like this in VS code make sure you're adding a SQL CARBON EDIT

Reply:
I tried to do the thing you did earlier but it did not work for me, it kept on restoring the grid even when it should not, I know this isn't right.